Fix tests

* Date you used Cookiecutter PyPackage: 6 April 2024
* Cookiecutter version used, if any: 2.6.0
* Python version, if any: 3.12.2
* Operating System: macOS 12.7.4

### Description

Describe what you were trying to get done. Tell us what happened, what went wrong, and what you expected to happen.

I was trying to run the tests, and I got failures.

### What I Did

```
% pytest
================================================ test session starts ================================================
platform darwin -- Python 3.12.2, pytest-8.1.1, pluggy-1.4.0
rootdir: /Users/drg/projects/cookiecutter-pypackage
configfile: pytest.ini
testpaths: tests/
plugins: cookies-0.7.0
collected 19 items                                                                                                  

tests/test_bake_project.py ..FFF........FFFF..                                                                [100%]

===================================================== FAILURES ======================================================
______________________________________________ test_bake_and_run_tests ______________________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x106c65b20>

    def test_bake_and_run_tests(cookies):
        with bake_in_temp_dir(cookies) as result:
            assert result.project.isdir()
>           run_inside_dir('python setup.py test', str(result.project)) == 0

tests/test_bake_project.py:91: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_bake_project.py:51: in run_inside_dir
    return subprocess.check_call(shlex.split(command))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

popenargs = (['python', 'setup.py', 'test'],), kwargs = {}, retcode = 1, cmd = ['python', 'setup.py', 'test']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['python', 'setup.py', 'test']' returned non-zero exit status 1.

/usr/local/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py:413: CalledProcessError
----------------------------------------------- Captured stderr call ------------------------------------------------
Traceback (most recent call last):
  File "/private/var/folders/zk/zl0j90056hqgwvnzc_csh0pm0000gn/T/pytest-of-drg/pytest-2/test_bake_and_run_tests0/cookies/bake00/python_boilerplate/setup.py", line 5, in <module>
    from setuptools import setup, find_packages
ModuleNotFoundError: No module named 'setuptools'
_____________________________________ test_bake_withspecialchars_and_run_tests ______________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x1071fdee0>

    def test_bake_withspecialchars_and_run_tests(cookies):
        """Ensure that a `full_name` with double quotes does not break setup.py"""
        with bake_in_temp_dir(
            cookies,
            extra_context={'full_name': 'name "quote" name'}
        ) as result:
            assert result.project.isdir()
>           run_inside_dir('python setup.py test', str(result.project)) == 0

tests/test_bake_project.py:102: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_bake_project.py:51: in run_inside_dir
    return subprocess.check_call(shlex.split(command))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

popenargs = (['python', 'setup.py', 'test'],), kwargs = {}, retcode = 1, cmd = ['python', 'setup.py', 'test']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['python', 'setup.py', 'test']' returned non-zero exit status 1.

/usr/local/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py:413: CalledProcessError
----------------------------------------------- Captured stderr call ------------------------------------------------
Traceback (most recent call last):
  File "/private/var/folders/zk/zl0j90056hqgwvnzc_csh0pm0000gn/T/pytest-of-drg/pytest-2/test_bake_withspecialchars_and0/cookies/bake00/python_boilerplate/setup.py", line 5, in <module>
    from setuptools import setup, find_packages
ModuleNotFoundError: No module named 'setuptools'
______________________________________ test_bake_with_apostrophe_and_run_tests ______________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x1071fe720>

    def test_bake_with_apostrophe_and_run_tests(cookies):
        """Ensure that a `full_name` with apostrophes does not break setup.py"""
        with bake_in_temp_dir(
            cookies,
            extra_context={'full_name': "O'connor"}
        ) as result:
            assert result.project.isdir()
>           run_inside_dir('python setup.py test', str(result.project)) == 0

tests/test_bake_project.py:112: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_bake_project.py:51: in run_inside_dir
    return subprocess.check_call(shlex.split(command))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

popenargs = (['python', 'setup.py', 'test'],), kwargs = {}, retcode = 1, cmd = ['python', 'setup.py', 'test']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['python', 'setup.py', 'test']' returned non-zero exit status 1.

/usr/local/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py:413: CalledProcessError
----------------------------------------------- Captured stderr call ------------------------------------------------
Traceback (most recent call last):
  File "/private/var/folders/zk/zl0j90056hqgwvnzc_csh0pm0000gn/T/pytest-of-drg/pytest-2/test_bake_with_apostrophe_and_0/cookies/bake00/python_boilerplate/setup.py", line 5, in <module>
    from setuptools import setup, find_packages
ModuleNotFoundError: No module named 'setuptools'
________________________________________ test_bake_with_console_script_files ________________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x10728c350>

    def test_bake_with_console_script_files(cookies):
        context = {'command_line_interface': 'click'}
        result = cookies.bake(extra_context=context)
        project_path, project_slug, project_dir = project_info(result)
>       found_project_files = os.listdir(project_dir)
E       FileNotFoundError: [Errno 2] No such file or directory: 'None/None'

tests/test_bake_project.py:275: FileNotFoundError
___________________________________ test_bake_with_argparse_console_script_files ____________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x10743b8f0>

    def test_bake_with_argparse_console_script_files(cookies):
        context = {'command_line_interface': 'argparse'}
        result = cookies.bake(extra_context=context)
        project_path, project_slug, project_dir = project_info(result)
>       found_project_files = os.listdir(project_dir)
E       FileNotFoundError: [Errno 2] No such file or directory: 'None/None'

tests/test_bake_project.py:287: FileNotFoundError
_________________________________________ test_bake_with_console_script_cli _________________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x107438440>

    def test_bake_with_console_script_cli(cookies):
        context = {'command_line_interface': 'click'}
        result = cookies.bake(extra_context=context)
        project_path, project_slug, project_dir = project_info(result)
        module_path = os.path.join(project_dir, 'cli.py')
        module_name = '.'.join([project_slug, 'cli'])
        spec = importlib.util.spec_from_file_location(module_name, module_path)
        cli = importlib.util.module_from_spec(spec)
>       spec.loader.exec_module(cli)

tests/test_bake_project.py:303: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
<frozen importlib._bootstrap_external>:991: in exec_module
    ???
<frozen importlib._bootstrap_external>:1128: in get_code
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_frozen_importlib_external.SourceFileLoader object at 0x10743aa20>
path = '/Users/drg/projects/cookiecutter-pypackage/None/None/cli.py'

>   ???
E   FileNotFoundError: [Errno 2] No such file or directory: '/Users/drg/projects/cookiecutter-pypackage/None/None/cli.py'

<frozen importlib._bootstrap_external>:1186: FileNotFoundError
____________________________________ test_bake_with_argparse_console_script_cli _____________________________________

cookies = <pytest_cookies.plugin.Cookies object at 0x1074387d0>

    def test_bake_with_argparse_console_script_cli(cookies):
        context = {'command_line_interface': 'argparse'}
        result = cookies.bake(extra_context=context)
        project_path, project_slug, project_dir = project_info(result)
        module_path = os.path.join(project_dir, 'cli.py')
        module_name = '.'.join([project_slug, 'cli'])
        spec = importlib.util.spec_from_file_location(module_name, module_path)
        cli = importlib.util.module_from_spec(spec)
>       spec.loader.exec_module(cli)

tests/test_bake_project.py:324: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
<frozen importlib._bootstrap_external>:991: in exec_module
    ???
<frozen importlib._bootstrap_external>:1128: in get_code
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_frozen_importlib_external.SourceFileLoader object at 0x10743bc80>
path = '/Users/drg/projects/cookiecutter-pypackage/None/None/cli.py'

>   ???
E   FileNotFoundError: [Errno 2] No such file or directory: '/Users/drg/projects/cookiecutter-pypackage/None/None/cli.py'

<frozen importlib._bootstrap_external>:1186: FileNotFoundError
================================================= warnings summary ==================================================
tests/test_bake_project.py: 61 warnings
  /Users/drg/.virtualenvs/cc-pp-env/lib/python3.12/site-packages/pytest_cookies/plugin.py:33: DeprecationWarning: project is deprecated and will be removed in a future release, please use project_path instead.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================== short test summary info ==============================================
FAILED tests/test_bake_project.py::test_bake_and_run_tests - subprocess.CalledProcessError: Command '['python', 'setup.py', 'test']' returned non-zero exit status 1.
FAILED tests/test_bake_project.py::test_bake_withspecialchars_and_run_tests - subprocess.CalledProcessError: Command '['python', 'setup.py', 'test']' returned non-zero exit status 1.
FAILED tests/test_bake_project.py::test_bake_with_apostrophe_and_run_tests - subprocess.CalledProcessError: Command '['python', 'setup.py', 'test']' returned non-zero exit status 1.
FAILED tests/test_bake_project.py::test_bake_with_console_script_files - FileNotFoundError: [Errno 2] No such file or directory: 'None/None'
FAILED tests/test_bake_project.py::test_bake_with_argparse_console_script_files - FileNotFoundError: [Errno 2] No such file or directory: 'None/None'
FAILED tests/test_bake_project.py::test_bake_with_console_script_cli - FileNotFoundError: [Errno 2] No such file or directory: '/Users/drg/projects/cookiecutter-pypackage/None/None/cl...
FAILED tests/test_bake_project.py::test_bake_with_argparse_console_script_cli - FileNotFoundError: [Errno 2] No such file or directory: '/Users/drg/projects/cookiecutter-pypackage/None/None/cl...
===================================== 7 failed, 12 passed, 61 warnings in 6.45s =====================================

```

--- a/tests/test_bake_project.py
+++ b/tests/test_bake_project.py
@@ -66,6 +66,9 @@ def test_year_compute_in_license_file(cookies):
 
 def project_info(result):
     """Get toplevel dir, project_slug, and project dir from baked cookies"""
+    assert result.exception is None
+    assert result.project.isdir()
+
     project_path = str(result.project)
     project_slug = os.path.split(project_path)[-1]
     project_dir = os.path.join(project_path, project_slug)
@@ -269,19 +272,7 @@ def test_bake_with_no_console_script(cookies):
 
 
 def test_bake_with_console_script_files(cookies):
-    context = {'command_line_interface': 'click'}
-    result = cookies.bake(extra_context=context)
-    project_path, project_slug, project_dir = project_info(result)
-    found_project_files = os.listdir(project_dir)
-    assert "cli.py" in found_project_files
-
-    setup_path = os.path.join(project_path, 'setup.py')
-    with open(setup_path, 'r') as setup_file:
-        assert 'entry_points' in setup_file.read()
-
-
-def test_bake_with_argparse_console_script_files(cookies):
-    context = {'command_line_interface': 'argparse'}
+    context = {'command_line_interface': 'Click'}
     result = cookies.bake(extra_context=context)
     project_path, project_slug, project_dir = project_info(result)
     found_project_files = os.listdir(project_dir)
@@ -293,28 +284,7 @@ def test_bake_with_argparse_console_script_files(cookies):
 
 
 def test_bake_with_console_script_cli(cookies):
-    context = {'command_line_interface': 'click'}
-    result = cookies.bake(extra_context=context)
-    project_path, project_slug, project_dir = project_info(result)
-    module_path = os.path.join(project_dir, 'cli.py')
-    module_name = '.'.join([project_slug, 'cli'])
-    spec = importlib.util.spec_from_file_location(module_name, module_path)
-    cli = importlib.util.module_from_spec(spec)
-    spec.loader.exec_module(cli)
-    runner = CliRunner()
-    noarg_result = runner.invoke(cli.main)
-    assert noarg_result.exit_code == 0
-    noarg_output = ' '.join([
-        'Replace this message by putting your code into',
-        project_slug])
-    assert noarg_output in noarg_result.output
-    help_result = runner.invoke(cli.main, ['--help'])
-    assert help_result.exit_code == 0
-    assert 'Show this message' in help_result.output
-
-
-def test_bake_with_argparse_console_script_cli(cookies):
-    context = {'command_line_interface': 'argparse'}
+    context = {'command_line_interface': 'Click'}
     result = cookies.bake(extra_context=context)
     project_path, project_slug, project_dir = project_info(result)
     module_path = os.path.join(project_dir, 'cli.py')

